torch/_dynamo/variables/functions.py

```diff
@@ -338,6 +338,10 @@ def call_function(
                 return self.obj.call_method(
                     tx, self.fn.__name__, args, kwargs, constant=self.is_constant
                 )
+        elif variables.TorchCtxManagerClassVariable.is_matching_cls(self.fn):
+            return variables.TorchCtxManagerClassVariable(self.fn).call_function(
+                tx, (self.obj, *args), kwargs
+            )
```

Comment:
I'd suggest to narrow down to the exact case you want to hijack, otherwise, you probably make other ctx manager classes fall into this bucket, which is not correct. I think this is probably the reason of the test failure at ```test_torch_profiler```.

Reply:
@yanboliang you are right - I just changed this to make it more specific to FSDP use_training_state